### PR TITLE
Pu/mergeall+refactor

### PR DESCRIPTION
**Pull request by izulin**
### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1.
2.
3.

### Checklist:

- [ ] My code follows the code style of this project,
- [ ] My change requires a change to the documentation,
- [ ] I described the modification in the CHANGELOG.md file.

### CODE REVIEW

File: src/CrudOperations.ts [hunk 1]
```diff
@@ -249,7 +249,7 @@ export class CrudOperations {
     } else {
       for (let i = 0; i < cellContents.length; i++) {
         if (!(cellContents[i] instanceof Array)) {
-          throw new InvalidArgumentsError('an array of arrays or a raw cell value.')
+          throw new InvalidArgumentsError('an array of arrays or a data cell value.')
```
Comment (izulin) [inline src/CrudOperations.ts:252]:
```suggestion
          throw new InvalidArgumentsError('an array of arrays or a raw cell value.')
```

File: docs/guide/custom-functions.md [hunk 2]
```diff
@@ -125,12 +125,12 @@ export const myTranslations = {
 For the simplicity of a basic example, you will not pass any
 arguments. However, this method imposes a particular structure to
 be used; there are two optional arguments, `ast` and
-`state`, and the function must return the results of
+`formulaAddress`, and the function must return the results of
```
Comment (izulin) [inline docs/guide/custom-functions.md:128]:
```suggestion
`state`, and the function must return the results of
```

Comment (izulin) [inline docs/guide/custom-functions.md]:
```suggestion
`state`, and the function must return the results of
```

File: docs/guide/custom-functions.md [hunk 3]
```diff
@@ -130,7 +130,7 @@ the calculations.
 
 ```javascript
 // arguments here are displayed just to show the structure
-public hyper(ast, state) {
+public hyper(ast, formulaAddress) {
```
Comment (izulin) [inline docs/guide/custom-functions.md:133]:
```suggestion
public hyper(ast, state) {
```

File: docs/guide/custom-functions.md [hunk 4]
```diff
@@ -152,7 +152,7 @@ export class CountHF extends FunctionPlugin {
     }
   };
 
-  public hyper(ast, state) {
+  public hyper(ast, formulaAddress) {
```
Comment (izulin) [inline docs/guide/custom-functions.md:155]:
```suggestion
  public hyper(ast, state) {
```

File: src/HyperFormula.ts [hunk 5]
```diff
@@ -3155,6 +3155,7 @@ export class HyperFormula implements TypedEmitter {
    *
    * @throws [[NamedExpressionNameIsAlreadyTakenError]] when the named expression name is not available.
    * @throws [[NamedExpressionNameIsInvalidError]] when the named expression name is not valid
+   * @throws [[MatrixFormulasNotSupportedError]] when the named expression formula is a Matrix formula
```
Comment (wojciechczerniak) [inline src/HyperFormula.ts:3158]: Those were removed, but are back?

**Review by wojciechczerniak — APPROVED** [review]

File: src/HyperFormula.ts [hunk 6]
```diff
@@ -3293,6 +3293,7 @@ export class HyperFormula implements TypedEmitter {
    * @throws [[ExpectedValueOfTypeError]] if any of its basic type argument is of wrong type
    * @throws [[NamedExpressionNameIsAlreadyTakenError]] when the named expression name is not available.
    * @throws [[NamedExpressionNameIsInvalidError]] when the named expression name is not valid
+   * @throws [[MatrixFormulasNotSupportedError]] when the named expression formula is a Matrix formula
```
Comment (izulin) [inline src/HyperFormula.ts:3296]:
```suggestion
```

File: src/HyperFormula.ts [hunk 7]
```diff
@@ -3506,6 +3507,7 @@ export class HyperFormula implements TypedEmitter {
    * @throws [[ExpectedValueOfTypeError]] if any of its basic type argument is of wrong type
    * @throws [[NamedExpressionDoesNotExistError]] when the given expression does not exist.
    * @throws [[NoSheetWithNameError]] when the given sheet name does not exists
+   * @throws [[MatrixFormulasNotSupportedError]] when the named expression formula is a Matrix formula
```
Comment (izulin) [inline src/HyperFormula.ts:3510]:
```suggestion
```

**Pull request merged**